python/custreamz/custreamz/tests/conftest.py

```diff
@@ -1,22 +1,29 @@
 # Copyright (c) 2020, NVIDIA CORPORATION.
+import socket
+
 import pytest
-from confluent_kafka.admin import AdminClient
 
 from custreamz import kafka
 
-# Test for the prescence of a running kafka broker
-kafka_available = True
-admin_client = AdminClient(
-    {"bootstrap.servers": "localhost:9092", "socket.timeout.ms": 1000}
-)
-topics = admin_client.list_topics().topics
-
-if not topics:
-    kafka_available = False
-
 
 @pytest.fixture(scope="session")
 def kafka_client():
+
+    # Check for the existance of a kafka broker
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        result = s.connect_ex(("localhost", 9092))
+        if result == 0:
+            print("Kafka service is running")
+        else:
+            print("Kafka service is NOT running")
+        s.shutdown(2)
+        s.close()
+    except Exception:
+        pytest.skip(
+            "A running Kafka instance must be" " available to run these tests"
+        )
```

Comment:
Instead of doing this what if we just used a try except surrounding the `kafka.Consumer(kafka_configs)` call? I.E.
```
try:
    consumer = kafka.Consumer(kafka_configs)
except ConnectionError:  #This is a made up exception type
    pytest.skip("...")

return consumer
```

Reply:
This proves difficult because if a Kafka instance is not running the call to attempt to create the Kafka consumer will infinitely attempt to reconnect, therefore never throwing an exception or returning. It does this because with broker rebalancing it allows for servers to be taken down or reassigned so it assumes you always provide the appropriate connection details and the server just isn't back online yet. 

It is possible to add an "event_cb" and try to parse that information out and throw an exception but that seems really error prone since indeed the lose of the connection could be something valid like a busy server not responding but a cluster being reconfigured.

Basically checking by socket is the only guaranteed way I can think of to ensure its just "down" and not something else expected going on.

Reply:
Sounds good to me